### PR TITLE
Commenting on the use of X_test as val. See #3196.

### PR DESCRIPTION
**Pull request by evanmiltenburg**
At least we can put our knowledge from discussing the issue in the comments. This will prevent future issues based on the same thing.

### CODE REVIEW

**Comment by fchollet** [comment]
The obsession about train/validation/test is intriguing to me. It seems to be indicative of a lack of understanding of the purpose of a validation set. For people in this situation, I recommend doing Kaggle competitions, where rigorous model validation is essential. This will allow you to build solid understanding of model validation, information leaks, and overfitting.
